tool/tsh/kube.go

```diff
@@ -638,6 +658,10 @@ func (c *kubeCredentialsCommand) run(cf *CLIConf) error {
 
 	ctx, span := tc.Tracer.Start(cf.Context, "tsh.kubeCredentials/RetryWithRelogin")
 	err = client.RetryWithRelogin(ctx, tc, func() error {
+		if err := c.checkLocalProxyRequirement(tc.TLSRoutingConnUpgradeRequired); err != nil {
```

Comment:
Can we call the `c.checkLocalProxyRequirement(tc.TLSRoutingConnUpgradeRequired)` earlier ?